### ISSUE TITLE
[PRODUCT]: Backfill NCEP Stage 4 Data

### ISSUE THREAD

**Issue by msweier** [USACE/cumulus#423]
Stage 4 data goes back about 2 years. Request to backfill as far as we can go (early 2000s).

**Comment by msweier** [comment]
@Enovotny, @adamscarberry, @jbkolze how would this work? A dag with a long look back that is only turned on manually in the production environment? This would be helpful for many districts who have legacy grids on their servers which won't be available when we move on and hopefully prevent the need to load most legacy precip data.

**Comment by adamscarberry** [comment]
@msweier Yes, create a copy dag and add a `backfill` suffix to the name.  We can always remove it later when done.

One thing to keep in mind is the speed of the load.  Need to throttle it to maybe one or two tasks at a time.  This also serves to not abuse the source server.

Let's say you have a time window from `2000` - `2020`.  The way I've done it before is to move the `end_date` a little at a time (via commits and push in github) starting with `2000`

Maybe start with no more than 5 years at a time.  We can always turn off the dag to allow a cool down.

We just don't want to impact other production geoprocessing by filling the queue with too many backload messages.

Might want to just wait and do this is in the new prod env, but we could give it a test run in dev.

**Comment by msweier** [comment]
Thanks @adamscarberry, since this is a CONUS product I suspect it may be pretty intensive. I'll check w/ HEC before I try anything....I see this is flagged as a low priority.

**Comment by msweier** [comment]
It looks like the production link only goes back a two weeks. To backfill POR we would need to go here:
https://rda.ucar.edu/datasets/d507005/dataaccess/#stage42002

As of the time of this post, it looks like the links are fairly stable and you can download a year of data with something like https://data-osdf.rda.ucar.edu/ncar/rda/d507005/stage4/stage4.200201.tar. which contains a nest of zipped files resolving to grib with a name like ST4.2002010100.01h.

Understood we would want to hold on this until production fully migrated.

**Comment by msweier** [comment]
Since the Stage IV data changes formats throughout its history, and since the AORC product incorporates the Stage IV data and more, I think we are better off trying to get AORC in cumulus instead. https://github.com/USACE/cumulus/issues/377
